pkg/live/inventoryrg.go

```diff
@@ -249,78 +308,432 @@ func (icm *InventoryResourceGroup) GetObject() (*unstructured.Unstructured, erro
+	totalStored := 0
+
+	for totalStored < len(icm.objMetas) {
+		var items []interface{}
+
+		switch ResourceCount {
+		case DisableSharding:
+			for _, objMeta := range icm.objMetas {
+				items = append(items, idToUnstructuedResources(objMeta))
+			}
+		default: // Split managed objects into current ResourceGroup.
+			for partitionCount := 0; partitionCount < ResourceCount && totalStored+partitionCount < len(icm.objMetas); partitionCount++ {
+				objMeta := icm.objMetas[totalStored+partitionCount]
+				items = append(items, idToUnstructuedResources(objMeta))
+			}
+		}
+
+		var rg *unstructured.Unstructured
+		switch totalStored {
+		case 0: // Initial RG in series.
+			rg = icm.inv
+			rg.SetResourceVersion("")
+		default:
+			rg = icm.inv.DeepCopy()
+			rg.SetResourceVersion("")
+			rg.SetName(fmt.Sprintf(shardedRGNameFmt, rg.GetName(), len(rg.GetName()), totalStored/ResourceCount))
+		}
+
+		addShardedLabel(rg, icm.Name())
+
+		err := unstructured.SetNestedSlice(rg.Object, items, "spec", "resources")
+		if err != nil {
+			return err
+		}
+
+		_, err = namespacedClient.Create(context.TODO(), rg, metav1.CreateOptions{})
+		if err != nil {
+			return err
+		}
+
+		totalStored += len(items)
+	}
+
+	return nil
 }
 
-func (icm *InventoryResourceGroup) getNamespacedClient(dc dynamic.Interface, mapper meta.RESTMapper) (*unstructured.Unstructured, dynamic.ResourceInterface, error) {
+// getNamespacedClient returns a namespaced client for interacting with ResourceGroups on cluster.
+func (icm *InventoryResourceGroup) getNamespacedClient(dc dynamic.Interface, mapper meta.RESTMapper) (dynamic.ResourceInterface, error) {
 	invInfo, err := icm.GetObject()
 	if err != nil {
-		return nil, nil, err
+		return nil, err
 	}
 	if invInfo == nil {
-		return nil, nil, fmt.Errorf("attempting to create a nil inventory object")
+		return nil, errors.New("attempting to interact with a nil inventory object")
 	}
 
 	mapping, err := mapper.RESTMapping(invInfo.GroupVersionKind().GroupKind(), invInfo.GroupVersionKind().Version)
 	if err != nil {
-		return nil, nil, err
+		return nil, fmt.Errorf("unable to create rest mapper for %s ResourceGroup: %w", invInfo.GroupVersionKind().Version, err)
 	}
 
 	// Create client to interact with cluster.
 	namespacedClient := dc.Resource(mapping.Resource).Namespace(invInfo.GetNamespace())
 
-	return invInfo, namespacedClient, nil
+	return namespacedClient, nil
+}
+
+// idFromFieldInterface serializes a nested field of stored GKNN resources within an unstructured ResourceGroup object,
+// to an ObjectMetadata.
+func idFromFieldInterface(fields interface{}) (object.ObjMetadata, error) {
+	item, ok := fields.(map[string]interface{})
+	if !ok {
+		return object.ObjMetadata{}, fmt.Errorf("unable to cast field to the required map type for: %v", fields)
+	}
+
+	id := object.ObjMetadata{
+		GroupKind: schema.GroupKind{
+			Group: item["group"].(string),
```

Comment:
These with panic if not strings.
I suspect you want to use NestedString to retrieve them and return any errors.
It shouldn't error if it came from the CRD with the right scheme, but you never know.